From ajit.khaparde@emulex.com  Thu Jun 23 15:40:25 2011
From: Ajit Khaparde <ajit.khaparde@emulex.com>
Date: Wed, 6 Apr 2011 05:53:13 +0000
Subject: [PATCH] be2net: Fix a potential crash during shutdown.
Git-commit: 2d5d4154650459b61a8e7097d186a89d89dab8ed
Patch-mainline: v2.6.39-rc3
References: FATE#311448, bnc#697255

adapter could remain uninitialized if probe fails for some reason.
A null pointer access could cause a crash if be_shutdown
is called after that.

Signed-off-by: Ajit Khaparde <ajit.khaparde@emulex.com>
Signed-off-by: David S. Miller <davem@davemloft.net>

Acked-by: Jeff Mahoney <jeffm@suse.com>
---
 drivers/net/benet/be_main.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)
--- a/drivers/net/benet/be_main.c
+++ b/drivers/net/benet/be_main.c
@@ -3061,7 +3061,9 @@ static int be_resume(struct pci_dev *pde
 static void be_shutdown(struct pci_dev *pdev)
 {
 	struct be_adapter *adapter = pci_get_drvdata(pdev);
-	struct net_device *netdev =  adapter->netdev;
+
+	if (!adapter)
+		return;
 
 	cancel_delayed_work_sync(&adapter->work);
 
